annotationframeworkclient/annotationengine.py

```diff
@@ -46,94 +69,103 @@ def server_address(self):
     @server_address.setter
     def server_address(self, value):
         self._server_address = value
-        self._default_url_mapping['ae_server_address'] = value
-
+        self._default_url_mapping["ae_server_address"] = value
 
     @property
     def default_url_mapping(self):
         return self._default_url_mapping.copy()
 
-
     @property
-    def cg_server_address(self):        
+    def cg_server_address(self):
         return self._server_address
 
     @cg_server_address.setter
     def cg_server_address(self, value):
         self._cg_server_address = value
-        self._default_url_mapping['cg_server_address'] = value
-
-    def infoserviceclient(self):
-        if self._infoserviceclient is None:
-            self._infoserviceclient = infoservice.InfoServiceClient(server_address=self.server_address, dataset_name=self.dataset_name)
-
-        return self._infoserviceclient
+        self._default_url_mapping["cg_server_address"] = value
 
     def get_datasets(self):
-        """ Returns existing datasets
+        """ Gets a list of datasets
 
-        :return: list
+        Returns
+        -------
+        list
+            List of dataset names for available datasets on the annotation engine
         """
         url = ae["datasets"].format_map(self.default_url_mapping)
         response = self.session.get(url)
-        assert(response.status_code == 200)
+        response.raise_for_status()
         return response.json()
 
     def get_tables(self, dataset_name=None):
-        """ Reads table infos
+        """ Gets a list of table names for a dataset
+
+        Parameters
+        ----------
+        dataset_name : str or None, optional
+            Name of the dataset, by default None. If None, uses the one specified in the client.
 
-        :param dataset_name: str
-        :return: dict
+        Returns
+        -------
+        list
+            List of table names
         """
         if dataset_name is None:
             dataset_name = self.dataset_name
         endpoint_mapping = self.default_url_mapping
-        endpoint_mapping['dataset_name'] = dataset_name
+        endpoint_mapping["dataset_name"] = dataset_name
         url = ae["table_names"].format_map(endpoint_mapping)
 
         response = self.session.get(url)
-        assert(response.status_code==200)
+        response.raise_for_status()
         return response.json()
 
     def create_table(self, table_name, schema_name, dataset_name=None):
```

Comment:
thinking forward.. what should we do for reference tables? should that be a different call or add an optional keyword argument for the name of the reference table and raise an error if you specify a reference schema but fail to provide a reference table?

Reply:
I think we should do this through a separate endpoint on the annotation engine side, and thus a separate function here. It will make the documentation much cleaner, as well.